### ISSUE TITLE
Support for an Apex class "outline view" pane for methods, properties, constructors, etc.

### ISSUE THREAD

**Issue by ImJohnMDaniel** [joeferraro/MavensMate-Atom#160]
It would be nice to have a "outline view" pane on the right side which shows a parsed view of the currently active Apex class.  Similar to the feature in the Force.com IDE also named "Outline View".

Is this feature on the roadmap?

**Comment by joeferraro** [comment]
Hiya, John! Thanks for the recommendation. I'm with you on this request, however, to do this properly, we really need the official Apex parser from Salesforce (gosh I feel like we've been waiting on this for a decade). I've been told that we'll get early access to it a bit later this year. 

The Apex grammar associated with this project works reasonably well with the Atom tokenizer but it's not perfect. We could start a baseline solution with our grammar, but I fear it will prompt a slew of issues complaining about deficiencies in things like getter/setter parsing. Then, we'll be scrambling to jury rig the grammar until we get the official parser from Salesforce. :-1:  Thoughts?

**Comment by yenkey** [comment]
Was  trying to find any plugins for sublime that support outline view and stumbled into this thread. Would love to see the outline view from sublime/mavenmate. @joeferraro just curious how the eclipse does it without apex parser. Does it use java's parser?
